src/main/java/org/folio/circulation/support/http/server/WebContext.java

```diff
@@ -19,9 +21,13 @@
 
 public class WebContext {
   private final RoutingContext routingContext;
+  private final Map<String, String> headers;
 
   public WebContext(RoutingContext routingContext) {
     this.routingContext = routingContext;
+
+    headers = routingContext.request().headers().entries().stream()
+      .collect(Collectors.toMap(entry -> entry.getKey().toLowerCase(), Map.Entry::getValue));
```

Comment:
Why does this need to be pre-generated rather than during the property getter?

Reply:
I'm not sure if headers can be changed via RoutingContext, they probably can't, but just to be safe I removed the field and moved this to the getter.